### PR TITLE
chore: add loading state profile metrics

### LINKED ISSUE #2605: "no data available" while the profiler is loading is confusing
> 1. while it’s still loading, it says “No data available.“. Probably should show a spinner?

when looking at the profiler the ui shows a "no data available" message depending on how we're interacting with the backend here we should be able to show a loading ui/message instead of a definite no data available.

### PR DESCRIPTION
**Pull request by hamidzr**
## Commentary (optional)

fixes #2605 
TODO can we do this? 

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Comment by apizzini** [comment]
React code looks good, but we have to involve @stoksc and @armandmcqueen here.

Displaying "no data" until system metrics data loads (and are not empty) was a requirement.

**Comment by stoksc** [comment]
For context @hamidzr , there may not always be timings metrics, and the API isn't aware (it just returns what the harness gives it, it does not know which framework we are on). So the goal was to avoid spinning forever on timings for keras.

**Comment by stoksc** [comment]
I see this is only for system metrics, not timings. This is fine, those should always be there.

**Review by stoksc — APPROVED** [review]

**Comment by hamidzr** [comment]
> I see this is only for system metrics, not timings. This is fine, those should always be there (so long as a conf with `profiling.enabled = false` still renders with a link to the docs not a spinner.

yep the ones that have it disabled  still get the following
image

**Comment by hamidzr** [comment]
I'm gonna rebase to avoid any potential conflict with the new header changes that landed after tests ran here.
